app/src/main/java/com/example/tasky/feature_agenda/presentation/agenda_screen/AgendaViewModel.kt

```diff
@@ -110,27 +111,31 @@ class AgendaViewModel @Inject constructor(
                 }
             }
 
-            is AgendaEvent.ToggleItemCreationMenu -> {
+            AgendaEvent.ToggleItemCreationMenu -> {
                 _state.update {
                     it.copy(isItemCreationMenuVisible = !state.value.isItemCreationMenuVisible)
```

Comment:
Make sure to use the state value the `update` block gives you. Instead of `!state.value.isItemCreationMenuVisible`, use `!it.isItemCreationMenuVisible`. Otherwise you might end up with race conditions